jdaviz/core/marks.py

```diff
@@ -59,11 +59,77 @@ def _update_counts(self, *args):
         self.right.text = [f'{oob_right} \u25b6' if oob_right > 0 else '']
 
 
-class BaseSpectrumVerticalLine(Lines, HubListener):
+class PluginMark():
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.xunit = None
+        self.yunit = None
+        # whether to update existing marks when global display units are changed
+        self.auto_update_units = True
+        self.hub.subscribe(self, GlobalDisplayUnitChanged,
+                           handler=self._on_global_display_unit_changed)
+
+        if not self.auto_update_units:
+            return
+        if self.xunit is None:
+            self.set_x_unit()
+        if self.yunit is None:
+            self.set_y_unit()
```

Comment:
```suggestion
        if self.auto_update_units:
            if self.xunit is None:
                self.set_x_unit()
            if self.yunit is None:
                self.set_y_unit()
```

I'm not sure if the if statement is even needed now here (I doubt the plugin changes the `auto_update_units` value before `__init__` finishes), but if it is needed this is slightly shorter.